projects/aca-content/src/lib/components/bulk-actions-dropdown/bulk-actions-dropdown.component.spec.ts

```diff
@@ -0,0 +1,157 @@
+/*!
+ * Copyright © 2005-2024 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { BulkActionsDropdownComponent } from './bulk-actions-dropdown.component';
+import { Store } from '@ngrx/store';
+import { AppStore } from '@alfresco/aca-shared/store';
+import { BehaviorSubject, of } from 'rxjs';
+import { By } from '@angular/platform-browser';
+import { ContentActionType } from '@alfresco/adf-extensions';
+import { AppTestingModule } from '../../testing/app-testing.module';
+import { TranslationService } from '@alfresco/adf-core';
+
+describe('BulkActionsDropdownComponent', () => {
+  let component: BulkActionsDropdownComponent;
+  let fixture: ComponentFixture<BulkActionsDropdownComponent>;
+  let store: Store<AppStore>;
+  let translationService: TranslationService;
+  let bulkFormField: HTMLElement;
+
+  const totalItemsMock$: BehaviorSubject<number> = new BehaviorSubject(0);
+
+  const getElement = (selector: string): HTMLElement | null => fixture.debugElement.query(By.css(`[data-automation-id="${selector}"]`)).nativeElement;
+
+  const getLabelText = (selector: string): string => getElement(selector).textContent.trim();
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [BulkActionsDropdownComponent, AppTestingModule]
+    }).compileComponents();
+
+    store = TestBed.inject(Store);
+    translationService = TestBed.inject(TranslationService);
+    spyOn(store, 'select').and.returnValue(totalItemsMock$);
+    spyOn(translationService, 'get').and.callFake((key) => of(key));
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(BulkActionsDropdownComponent);
```

Comment:
I guess maybe we should move this to a single beforeEach()?? Calling TestBed.configureTestingModule() after TestBed.createComponent() will [throw an error](https://v17.angular.io/guide/testing-components-basics#createcomponent). Having it in 2 seperate beforeEach blocks creates a possibility of that happening

Reply:
done